scripts.Rmd

```diff
@@ -0,0 +1,255 @@
+# `drake` and script-based workflows {#scripts}
+
+```{r, message = FALSE, warning = FALSE, echo = FALSE}
+knitr::opts_knit$set(root.dir = fs::dir_create(tempfile()))
+knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
+```
+
+```{r, message = FALSE, warning = FALSE, echo = FALSE}
+library(drake)
+library(glue)
+library(purrr)
+library(rlang)
+library(tidyverse)
+invisible(drake_example("script-based-workflows", overwrite = TRUE))
+tmp <- file.copy(
+  list.files("script-based-workflows/R/",
+             pattern = "*.R", full.names = TRUE),
+  ".",
+  recursive = TRUE)
+tmp <- file.copy("script-based-workflows/raw_data.xlsx", ".")
+tmp <- file.copy("script-based-workflows/report.Rmd", ".")
+dir.create("data")
+```
+
+`drake` has been built with the firm belief that the best workflows rely on approaching problems from a function perspective. This idea is supported in "Good enough practices in scientific computing", that approaching your project by writing modular code increases the end product.
+
+> The core realization in these practices is that being readable, reusable, and testable are all side effects of writing modular code, i.e., of building programs out of short, single-purpose functions with clearly-defined inputs and outputs 
+> 2017 Wilson et al.
+
+That being said, not all projects and workflows are written using this mantra and for others the lift would be too great to re-factor to a purely functional approach. 
+
+## Traditional and Legacy Workflows
+
+A common approach to creating workflows is to write steps of the analysis into scripts that perform specific tasks. By giving these scripts meaningful names, e.g. 01_data.R, 02_munge.R, 03_analysis.R,  one can surmise both their contents and the order in which these scripts should be run.
+
+It was acknowledged in the (What we left out)[https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005510#sec027] section that build tools to manage projects when code gets out of date are ideal, but can be substituted with a simple master script called `make.R` that reruns each of the scripts in order.
```

Comment:
- In the link, the square brackets go around the text, and the parentheses go around the hyperlink. 
- Maybe show a Makefile with the scripts.

Reply:
Hmm..., maybe an actual Makefile gets too technical.

Reply:
In fact, we can probably just skip the discussion of build tools and jump to `code_to_function()`. Sketch:

`code_to_function()` can create a plan-ready function from each numbered script.

```{r, eval = TRUE}
# R/functions.R
load_data <- code_to_function("01_data.R")
do_munge  <- code_to_function("02_munge.R")
generate_histogram <- code_to_function("03_histogram.R")
do_regression <- code_to_function("04_regression.R")
generate_report <- code_to_function("05_report.R")

print(load_data)
``` 

Next, we develop a plan on top of these functions.

Reply:
Maybe that last comment belongs in the section on overcoming technical debt.